### PR TITLE
Add unit tests on pkg/mesos

### PR DESCRIPTION
**Pull request by rgo3**

### CODE REVIEW

File: pkg/mesos/client_test.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+						Hostname: "10.0.0.23",
+					},
+				},
+			},
+		},
+	}
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "/mesos/api/v1", r.URL.String())
+		assert.Equal(t, "application/x-protobuf", r.Header.Get("Accept"))
```
Comment (bamarni) [inline pkg/mesos/client_test.go:151]: :+1: 

Maybe also `assert.Equal(t, "application/x-protobuf", r.Header.Get("Content-Type"))`

File: pkg/mesos/client_test.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "/mesos/api/v1", r.URL.String())
+		var expectedBuffer bytes.Buffer
+		err := json.NewEncoder(&expectedBuffer).Encode(expectedBody)
+		assert.NoError(t, err)
+		requestBody, err := ioutil.ReadAll(r.Body)
+		assert.NoError(t, err)
+		requestBuffer := bytes.NewBuffer(requestBody)
+		assert.Equal(t, &expectedBuffer, requestBuffer)
```
Comment (bamarni) [inline pkg/mesos/client_test.go]: maybe it'd have been simpler to just decode the request body and compare the 2 structs, but as you already have it like that we can keep it as is.

File: pkg/mesos/client_test.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+	const expectedAgentID = "9001"
+	expectedBody := master.Call{
+		Type: master.Call_MARK_AGENT_GONE,
+		MarkAgentGone: &master.Call_MarkAgentGone{
+			AgentID: mesos.AgentID{Value: expectedAgentID},
+		},
+	}
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "/mesos/api/v1", r.URL.String())
```
Comment (bamarni) [inline pkg/mesos/client_test.go:176]: could be nice to also assert on the request method being `POST` here, and asserting on `GET` in other tests.

**Review by armandgrillet — APPROVED** [review]

**Comment by armandgrillet** [comment]
`ok  	github.com/dcos/dcos-core-cli/pkg/mesos	1.169s	coverage: 74.1% of statements`
Nice!

**Review by bamarni — APPROVED** [review]
Awesome, we're getting so close to our 75% coverage threshold :smile:

**Pull request merged**